tutor/src/components/annotations/annotation.jsx

```diff
@@ -0,0 +1,353 @@
+    }
+  }
+
+  saveAnnotation() {
+    window.getSelection().empty();
+
+    return hypothesisStore.save(this.props.documentId, this.savedSelection, '')
+      .then((response) => {
+        const newEntry = dbEntryToPreload(response);
+        const newPreload = this.state.preload.slice();
+
+        newPreload.push(newEntry);
+        this.setState({
+          preload: newPreload
+        });
+        return newEntry;
+      });
+  }
+
+  deleteEntry() {
+    this.unhighlightEntry(this.state.activeHighlight);
+    hypothesisStore.delete(this.state.activeHighlight.savedId).then((response) => {
+      const oldIndex = this.state.preload.findIndex((e) => e.savedId === response.id);
+      const newPreload = this.state.preload.slice();
+
+      newPreload.splice(oldIndex, 1);
+      this.setState({
+        preload: newPreload
+      });
+    });
+    this.setState({
+      activeHighlight: null
+    });
+    window.getSelection().empty();
+  }
+
+  nextAnnotationInSortedList(entries) {
+    const nextIndex = 1 + entries.findIndex((e) => e.selection.start === this.state.activeHighlight.selection.start);
+
+    if (nextIndex < entries.length) {
+      this.setState({activeHighlight: entries[nextIndex]});
+      serializeSelection.restore(entries[nextIndex].selection, this.articleElement);
+    }
+  }
+
+  nextAnnotation() {
+    const entries = this.state.preload.sort((a, b) => a.selection.start - b.selection.start);
+
+    this.nextAnnotationInSortedList(entries);
+  }
+
+  previousAnnotation() {
+    // The trick is that the sort is reversed
+    const entries = this.state.preload.sort((a, b) => b.selection.start - a.selection.start);
+
+    this.nextAnnotationInSortedList(entries);
+  }
+
+  getParentRect(el) {
+    if (el) {
+      const wLeft = window.pageXOffset || document.documentElement.scrollLeft;
+      const wTop = window.pageYOffset || document.documentElement.scrollTop;
+      const parentRect = el.parentNode.getBoundingClientRect();
+
+      this.articleElement = el.parentNode;
+      this.parentRect = {
+        bottom: wTop + parentRect.bottom,
+        left: wLeft + parentRect.left,
+        right: wLeft + parentRect.right,
+        top: wTop + parentRect.top
+      }
+    }
+  }
+
+  render() {
+    return (
+      <div className="annotater" ref={this.getParentRect.bind(this)}>
+        <HighlightWidget
+          style={this.state.widgetStyle}
+          annotate={this.openAnnotator.bind(this)}
+          highlight={this.highlightAndClose.bind(this)}
+        />
+        <EditBox
+          show={this.state.activeHighlight ? 'open' : 'closed'}
+          annotation={this.state.activeHighlight ? this.state.activeHighlight.annotation : ''}
+          updateAnnotation={this.updateActiveAnnotation.bind(this)}
+          save={this.updateAnnotation.bind(this)}
+          delete={this.deleteEntry.bind(this)}
+          next={this.nextAnnotation.bind(this)}
+          previous={this.previousAnnotation.bind(this)}
+        />
```

Comment:
`binding` either by using `bind` or stabby funcs can cause duplicate rendering since the props of EditBox will always change.  https://ryanfunduk.com/articles/never-bind-in-render/

Instead it's better to bind the functions in `constructor` for plain React.  We usually use mobx `@action.bound` if it's an action or `core-decorators` `@autobind` decorators